### ISSUE TITLE
Web Interface

### ISSUE THREAD

**Issue by KillahB33** [Gkiokan/ps4-remote-pkg-sender#75]
I saw #45 and that it was closed completed but I don't see a way to host this like mentioned.

Is this still planned? Or is this staying purely an electron app?

**Comment by MafiaMac** [comment]
If you are using the windows port like just purely the exe then try and download the win.zip version this problem also happened to me so I downloaded that and it runs

**Comment by MafiaMac** [comment]
ALSO I FORGOT TO MENTION

Make sure you have Remote Package Installer on your Play Station open so it can run

**Comment by KillahB33** [comment]
I'm not sure how your comments relate in anyway to what I am asking.

@Gkiokan please let me know, am tempted to fork and take a stab at it but want to make sure I am not duplicating work.

**Comment by MafiaMac** [comment]
Ohh yeah my bad. I double checked what I am seeing is you are trying to run this on your web browser is that correct?

**Comment by KillahB33** [comment]
Yeah that's right, I see it's js based. My understanding is electron is used to take webapps and turn them into desktop apps. I am hoping that I can remove the electron piece so it will be a webapp and it will run on the nas so will reference the local filesystem.

**Comment by MafiaMac** [comment]
Alright so I did some basic research and here’s a article that should help remove electron from the project https://forum.endeavouros.com/t/how-to-completely-remove-electron-as-a-dependencie-of-element-desktop/37845 keep this noted this may break the app

**Comment by MafiaMac** [comment]
Let me know if you had done this before or not

**Comment by KillahB33** [comment]
I have not, I am up for the task but want to make sure I am not working on something that @Gkiokan is already working on.

**Comment by MafiaMac** [comment]
Just do it for fun even though there’s a possibility they might be working on it.

**Comment by Gkiokan** [comment]
The reason it is an electron app is that I can build for cross OS. That's the huge win that my app provides over most ones.

I would recommend a chrome/ff browser extension for the web interface part just because you need a bit of serving feature so the files has to be accessible on your network.

I wouldn't start by removing the electron part that doesn't sound correct to me. But inherit the js files and have a dedicated build command to have a web version that can be wrapped up to a chrome extension is that what I have planed already.

But before we dig into that, I have to finish the rpsv2 api server for the dynamic language and settings mapping for the upcoming host my files feature.

Honestly I will still be a couple weeks away before I can start doing the todo tasks on my app. I am restoring and pimping a synth (korg trinity) , you can see that on facebook if you are interested.

I will catch you up guys when I am ready. In the meantime you can get thr clue via the ps4 and Ps5 api and the server file which are the core core core functionality.

**Comment by MafiaMac** [comment]
Oh snap it’s @Gkiokan himself I want to give a short thank you for your contribution this software you made is amazing 🔥🔥🔥

**Comment by Gkiokan** [comment]
> I'm not sure how your comments relate in anyway to what I am asking.
> 
> @Gkiokan please let me know, am tempted to fork and take a stab at it but want to make sure I am not duplicating work.

As I said I'd like to build something that would have a dedicated build process without conflicting with the main app. But I don't have any idea yet how to build the structure. And I hate duplicate files, there is surely a nice way to do that.

If you can't hold yourself feel free and get your head around it. I am blocked for a couple weeks anyway. Get in touch with me on discord or telegram.

**Comment by KillahB33** [comment]
My plan would be to implement a docker container that lives on my nas, then have a web front end that is accessible from there.

Ok I'll fork and work through it, I'll message you on discord as well.

Really appreciate all you have done and your insights on this.

**Comment by Gkiokan** [comment]
> Oh snap it’s @Gkiokan himself I want to give a short thank you for your contribution this software you made is amazing 🔥🔥🔥

Thanks for the kind words. I try my best whenever I have time for it. But due the missing support I am not priorising it as top tier task. I mentioned that a couple times already. But I don't want this ads or pppups taht annoys everyone. If everyone would shoot at least 1 dollar donation over kofi I could finish so much stuff at once but meh life in Germany is expensive and the bills don't pay itself.

It's not an excuse, just a clarification about the reason that I unfortunately can't work 24/7 on it.
